integration/test/app/basic/simple_test.go

```diff
@@ -0,0 +1,201 @@
+		err = waitForUpdatedChartCR(ctx, create, "")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		chart, err := config.Host.G8sClient().ApplicationV1alpha1().Charts(namespace).Get(testAppReleaseName, v1.GetOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		if chart.Spec.TarballURL != tarballURL {
+			t.Fatalf("expected tarballURL: %#q got %#q", tarballURL, chart.Spec.TarballURL)
+		}
+		if chart.Labels[chartOperatorVersion] != "1.0.0" {
+			t.Fatalf("expected version label: %#q got %#q", "1.0.0", chart.Labels[chartOperatorVersion])
+		}
+		originalResourceVersion = chart.ObjectMeta.ResourceVersion
+	}
+
+	// Test update
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updating chart %#q", customResourceReleaseName))
+
+		sampleChart.App.Version = "1.0.1"
+
+		chartValues, err := chartvalues.NewAPIExtensionsAppE2E(sampleChart)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		chartInfo := release.NewStableChartInfo(customResourceReleaseName)
+		err = config.Release.Update(ctx, customResourceReleaseName, chartInfo, chartValues)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		err = config.Release.WaitForStatus(ctx, fmt.Sprintf("%s-%s", namespace, customResourceReleaseName), "DEPLOYED")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("updated chart %#q", customResourceReleaseName))
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking chart CR %#q is updated", testAppReleaseName))
+
+		tarballURL := "https://giantswarm.github.com/sample-catalog/test-app-1.0.1.tgz"
+		err = waitForUpdatedChartCR(ctx, update, originalResourceVersion)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		chart, err := config.Host.G8sClient().ApplicationV1alpha1().Charts(namespace).Get(testAppReleaseName, v1.GetOptions{})
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+		if chart.Spec.TarballURL != tarballURL {
+			t.Fatalf("expected tarballURL: %#v got %#v", tarballURL, chart.Spec.TarballURL)
+		}
+	}
+
+	// Test deletion
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleting chart %#q", customResourceReleaseName))
+
+		err := config.Release.Delete(ctx, customResourceReleaseName)
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		err = config.Release.WaitForStatus(ctx, fmt.Sprintf("%s-%s", namespace, customResourceReleaseName), "DELETED")
+		if err != nil {
+			t.Fatalf("expected %#v got %#v", nil, err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deleted chart %#q", customResourceReleaseName))
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking chart CR %#q is deleted", testAppReleaseName))
+
+		err = waitForUpdatedChartCR(ctx, delete, "")
```

Comment:
The log above talks about deletion but this is about updating. How does this work?

Reply:
It is trying to detect the changes from app CR, but can't find a name make more sense than this.

Reply:
Maybe the scope of the function is just wrong then or the primitive it provides is not really straight.

Reply:
waitForStatusChange is better?